### DISCUSSION TITLE
How to migrate from 5.0.1 to 5.1.0?

### DISCUSSION THREAD

**Discussion by Wizzel1** [zino-app/graphql-flutter#1129] [General]
I am testing my code with these stubs:

```dart
      when(
        () => graphQLHelper.query<dynamic>(
          query: GetNotesByAssessmentIdQuery.getNotesByAssessmentId,
          variables: <String, dynamic>{
            'id': assessmentId,
            'exerciseId': exerciseId,
            'userId': userId
          },
        ),
      ).thenAnswer(
        (_) => Future.value(
          QueryResult<dynamic>(
            data: jsonResponse,
            source: QueryResultSource.network,
            parserFn: (data) {},
          ),
        ),
      );
  ```

but I am getting an error when updating to 5.1.0. 

`The named parameter options is required.`

I have tried to use a `QueryOptions` object but that requires a document.

So what kind of "options" do I have to provide in this case?

**Comment by vincenzopalazzo** [comment]
You can create the query option in this way

```
final _options = QueryOptions(
          document: parseString(readRepositories),
          variables: <String, dynamic>{
            'nRepositories': 42,
          },
        );
```

In addition, now the option is required in the method call, and it is introduced from this PR https://github.com/zino-hofmann/graphql-flutter/commit/3dc626db6a17523a2398381d22236b110fbcf03b

I suggest using the query option, another example of this is inside this client https://github.com/LNOpenMetrics/lnmetrics.client/blob/main/lib/api/lnmetrics_api.dart

**Reply by Wizzel1** [reply]
Thank you!
